docs/changelog-fragments.d/805.deprecation.rst

```diff
@@ -0,0 +1,12 @@
+Deprecated :py:meth:`~cheroot.makefile.MakeFile` in favor of :py:class:`~cheroot.makefile.StreamReader`
+and :py:class:`~cheroot.makefile.StreamWriter`.
+
+The ``MakeFile`` factory function is replaced with direct class instantiation
```

Comment:
IIRC, this should let Sphinx check that the referenced object is correct without actually adding a hyperlink:
```suggestion
The :py:func:`!MakeFile` factory function is replaced with direct class instantiation
```

Reply:
*(the `!` idea is borrowed from CPython Style Guide, FTR: https://devguide.python.org/documentation/style-guide/#links)*